File: spring-bootstrap-gradle-profiles-docker/src/main/java/com/marcskow/spring/person/PersonRepository.java
package com.marcskow.spring.person;

import com.marcskow.spring.person.model.Person;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
interface PersonRepository extends CrudRepository<Person, Long> {
    Page<Person> findAll(Pageable pageable);
}

File: spring-bootstrap-complex-criteria-api/src/main/kotlin/com/marcskow/complex/criteria/api/person/PersonRepository.kt
package com.marcskow.complex.criteria.api.person

import com.marcskow.complex.criteria.api.person.model.Person
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface PersonRepository : CrudRepository<Person, Long> {
    fun findAll(pageable: Pageable): Page<Person>
}
